### ISSUE TITLE
kserve Python SDK installation fails on ray dependency resolution

### ISSUE THREAD

**Issue by elliottmoos** [kserve/kserve#2486]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I typically use poetry to manage project dependencies, but I tried installing the kserve Python SDK with both poetry and pip and got the same result for both:

Run `poetry add kserve`
```bash
 • Installing ray (1.10.0): Failed

  RuntimeError

  Unable to find installation candidates for ray (1.10.0)

  at ~/.poetry/lib/poetry/installation/chooser.py:72 in choose_for
       68│ 
       69│             links.append(link)
       70│ 
       71│         if not links:
    →  72│             raise RuntimeError(
       73│                 "Unable to find installation candidates for {}".format(package)
       74│             )
       75│ 
       76│         # Get the best link
```

Run `pip install kserve==0.9.0`
```bash
(.venv) bash-5.1$ pip install kserve==0.9.0
Collecting kserve==0.9.0
  Using cached kserve-0.9.0-py3-none-any.whl (304 kB)
Requirement already satisfied: setuptools>=59.5.0 in ./.venv/lib/python3.10/site-packages (from kserve==0.9.0) (60.6.0)
Collecting argparse>=1.4.0
  Using cached argparse-1.4.0-py2.py3-none-any.whl (23 kB)
Requirement already satisfied: azure-storage-file-share==12.7.0 in ./.venv/lib/python3.10/site-packages (from kserve==0.9.0) (12.7.0)
Requirement already satisfied: numpy~=1.21.5 in ./.venv/lib/python3.10/site-packages (from kserve==0.9.0) (1.21.6)
ERROR: Could not find a version that satisfies the requirement ray[serve]==1.10.0 (from kserve) (from versions: 1.13.0, 2.0.0rc0, 2.0.0rc1, 2.0.0, 2.0.1)
ERROR: No matching distribution found for ray[serve]==1.10.0
```

**What did you expect to happen:**
To successfully install the kserve Python SDK

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KFServing Version:
- Kubeflow version:
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by blakehawkins** [comment]
I think this might be related https://github.com/pypa/pip/issues/5402

**Comment by neggert** [comment]
Same here. Seems like versions of ray older than 1.13.0 have vanished from PyPI.

**Comment by neggert** [comment]
Aha. Think I figured it out. I was running on Python 3.10 and Ray didn't support 3.10 until v1.13. Dropping back to Python 3.9 solved it for me.

**Comment by gmuslia** [comment]
Issue still persists for me, I keep getting the following:

### 1. When I specify the version of kserve in the requirements.txt (using python:3.9)

10.77      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 10.8 MB/s eta 0:00:00
10.86 ERROR: Could not find a version that satisfies the requirement ray[serve]==1.10.0 (from kserve) (from versions: none)
10.86 ERROR: No matching distribution found for ray[serve]==1.10.0
11.01 WARNING: You are using pip version 22.0.4; however, version 22.3.1 is available.
11.01 You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.

### 2. When I do not specify the kserve version in the requirements.txt (using python:3.9)

23.43      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.8/7.8 MB 8.5 MB/s eta 0:00:00
23.44 ERROR: Cannot install -r ./requirements.txt (line 1) because these package versions have conflicting dependencies.
23.44
23.44 The conflict is caused by:
23.44     kserve 0.9.0 depends on ray[serve]==1.10.0
23.44     kserve 0.8.0.2 depends on ray[serve]==1.9.0
23.44     kserve 0.8.0.1 depends on ray[serve]==1.9.0
23.44     kserve 0.8.0 depends on ray[serve]==1.9.0
23.44     kserve 0.7.0 depends on ray[serve]==1.5.0
23.44
23.44 To fix this you could try to:
23.44 1. loosen the range of package versions you've specified
23.44 2. remove package versions to allow pip attempt to solve the dependency conflict


**Any ideas how to solve this?**

I am using MacBook M1 to build the docker image, the example I took is from here: **http://incredible.ai/kubernetes/2022/01/01/KServe-Custom-Model/** (except that I use python 3.9 and not 3.7)

**Comment by nachogarciam** [comment]
Hey @gmuslia, I solved the issue with this https://stackoverflow.com/questions/72239693/why-get-pip-install-error-when-run-docker-build-on-only-m1-mac, check it out

**Comment by yuzisun** [comment]
This should be fixed with KServe 0.10
https://github.com/kserve/kserve/pull/2410
